sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -188,30 +188,36 @@ case class GpuSlice(x: Expression, start: Expression, length: Expression)
 
   override def doColumnar(listCol: GpuColumnVector, startS: GpuScalar,
       lengthCol: GpuColumnVector): ColumnVector = {
-    // When start is null, return all nulls like the CPU does.
-    if (listCol.getRowCount == listCol.numNulls() || !startS.isValid) {
+    // When list, start or length is null, return all nulls like the CPU does.
+    if (listCol.getRowCount == listCol.numNulls() ||
+        lengthCol.getRowCount == lengthCol.numNulls() ||
+        !startS.isValid) {
       GpuColumnVector.columnVectorFromNull(listCol.getRowCount.toInt, dataType)
     } else {
       val list = listCol.getBase
       val start = startS.getValue.asInstanceOf[Int]
-      val length = lengthCol.getBase
       if (start == 0) {
         throw RapidsErrorUtils.unexpectedValueForStartInFunctionError(prettyName)
       }
-      withResource(length.min()) { minLen =>
-        if (minLen.isValid && minLen.getInt < 0) {
-          throw RapidsErrorUtils.unexpectedValueForLengthInFunctionError(prettyName,
-            minLen.getInt)
+      // keep null as original, or null it out if `list[i]`  is null
+      withResource(GpuDivModLike.mergeNulls(lengthCol.getBase, list)) { length =>
```

Comment:
nit: would it be better to move this to a different utility class? a collectionOperation is not really a DivModLike operation.